PurchasesCoreSwift/Networking/HTTPClient.swift

```diff
@@ -0,0 +1,306 @@
+    private let operationDispatcher: OperationDispatcher
+
+    @objc public init(systemInfo: SystemInfo, eTagManager: ETagManager, operationDispatcher: OperationDispatcher) {
+        let config = URLSessionConfiguration.ephemeral
+        config.httpMaximumConnectionsPerHost = 1
+        self.session = URLSession(configuration: config)
+        self.systemInfo = systemInfo
+        self.eTagManager = eTagManager
+        self.operationDispatcher = operationDispatcher
+    }
+
+    // TODO:(post-migration) remove and use only the private performRequest
+    @objc public func performRequest(_ httpMethod: String,
+                                     performSerially: Bool = false,
+                                     path: String,
+                                     requestBody: [String: Any]?,
+                                     headers authHeaders: [String: String],
+                                     completionHandler: ((Int, [AnyHashable: Any]?, Error?) -> Void)?) {
+        performRequest(httpMethod,
+                       performSerially: performSerially,
+                       path: path,
+                       requestBody: requestBody,
+                       authHeaders: authHeaders,
+                       retried: false,
+                       completionHandler: completionHandler)
+    }
+
+    @objc public func clearCaches() {
+        operationDispatcher.dispatchOnHTTPSerialQueue { [self] in
+            self.eTagManager.clearCaches()
+        }
+    }
+
+}
+
+private extension HTTPClient {
+
+    var defaultHeaders: [String: String] {
+        let observerMode = systemInfo.finishTransactions ? "false" : "true"
+        var headers: [String: String] = [
+            "content-type": "application/json",
+            "X-Version": SystemInfo.frameworkVersion,
+            "X-Platform": SystemInfo.platformHeader,
+            "X-Platform-Version": SystemInfo.systemVersion,
+            "X-Platform-Flavor": systemInfo.platformFlavor,
+            "X-Client-Version": SystemInfo.appVersion,
+            "X-Client-Build-Version": SystemInfo.buildVersion,
+            "X-Observer-Mode-Enabled": observerMode
+        ]
+
+        if let platformFlavorVersion = self.systemInfo.platformFlavorVersion {
+            headers["X-Platform-Flavor-Version"] = platformFlavorVersion
+        }
+
+        if let idfv = SystemInfo.identifierForVendor {
+            headers["X-Apple-Device-Identifier"] = idfv
+        }
+        return headers
+    }
+
+    func performRequest(_ httpMethod: String,
+                        performSerially: Bool = false,
+                        path: String,
+                        requestBody maybeRequestBody: [String: Any]?,
+                        authHeaders: [String: String],
+                        retried: Bool = false,
+                        completionHandler maybeCompletionHandler: ((Int, [AnyHashable: Any]?, Error?) -> Void)?) {
+        operationDispatcher.dispatchOnHTTPSerialQueue { [self] in
```

Comment:
same comment about scope here, also helps you catch `return` bugs since you can easily see if you're exiting a function or just a block but staying within the scope of the calling function.